### ISSUE TITLE
Circular import on latest commit

### ISSUE THREAD

**Issue by facutuesca** [sigstore/sigstore-python#989]
**Description**
```sh
# install latest commit
pip install "sigstore @ git+ssh://git@github.com/sigstore/sigstore-python@7e7cb04b6b5debb20050d2d857b2da88a18ca636"
python -c 'from sigstore.verify.models import Bundle'
```
Results in:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/facu/repos/test/lib/python3.12/site-packages/sigstore/verify/__init__.py", line 45, in <module>
    from sigstore.verify.models import Bundle
  File "/Users/facu/repos/test/lib/python3.12/site-packages/sigstore/verify/models.py", line 49, in <module>
    from sigstore.transparency import LogEntry, LogInclusionProof
  File "/Users/facu/repos/test/lib/python3.12/site-packages/sigstore/transparency.py", line 39, in <module>
    from sigstore._internal.rekor.checkpoint import verify_checkpoint
  File "/Users/facu/repos/test/lib/python3.12/site-packages/sigstore/_internal/rekor/__init__.py", line 28, in <module>
    from .client import RekorClient
  File "/Users/facu/repos/test/lib/python3.12/site-packages/sigstore/_internal/rekor/client.py", line 31, in <module>
    from sigstore.transparency import LogEntry
ImportError: cannot import name 'LogEntry' from partially initialized module 'sigstore.transparency' (most likely due to a circular import) (/Users/facu/repos/test/lib/python3.12/site-packages/sigstore/transparency.py)
```

**Version**
Latest commit (https://github.com/sigstore/sigstore-python/commit/7e7cb04b6b5debb20050d2d857b2da88a18ca636)

cc @woodruffw

**Comment by woodruffw** [comment]
This circularity is part of a longstanding mis-placement of types. I'm declaring module bankruptcy and placing this all under `sigstore.models`...

**Comment by woodruffw** [comment]
Fixed with #990.
